src/cfml/system/services/ServerService.cfc

```diff
@@ -1817,7 +1819,10 @@ component accessors="true" singleton {
 		} else if( serverInfo.AJPEnable ) {
 			portToCheck = serverInfo.AJPPort;
 		}
-		return !isPortAvailable( serverInfo.host, portToCheck );
+
+		lock name="server-status-check-#portToCheck#" type="exclusive"{
```

Comment:
@lmajano Does the thread safety issue only come into play when there are two servers using the same port and CommandBox is trying to bind to the same port twice in different threads?  I see you are incorporating the actual port itself into the lock name.